doc/examples/histology_resdnn.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+"""
+.. histology_resdnn:
+
+==================================================
+Local reconstruction using the Histological ResDNN
+==================================================
+
+A data-driven approach to modeling the non-linear mapping between observed
+DW-MRI signals and ground truth structures using sequential deep neural network
+regression with residual block deep neural network (ResDNN) [1, 2].
+
+Training was performed on two 3-D histology datasets of squirrel monkey brains
+and validated on a third. A second validation was performed on HCP datasets.
+"""
+
+import os
+
+from dipy.core.gradients import gradient_table
+from dipy.data import get_fnames, get_sphere
+from dipy.io.image import load_nifti, save_nifti
+from dipy.io.gradients import read_bvals_bvecs
+from dipy.nn.histo_resdnn import HistoResDNN
+from dipy.reconst.shm import sh_to_sf
+from dipy.segment.mask import median_otsu
+from dipy.viz import window, actor
+import numpy as np
+import scipy.ndimage as ndi
+
+
+# Disable oneDNN warning
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '2'
+
+"""
+This ResDNN model requires single-shell data with b0, the data is fetched
+and a gradient table is constructed from bvals/bvecs.
+"""
+
+# Fetch DWI and GTAB
+hardi_fname, hardi_bval_fname, hardi_bvec_fname = get_fnames('stanford_hardi')
+data, affine = load_nifti(hardi_fname)
+data = np.squeeze(data)
+bvals, bvecs = read_bvals_bvecs(hardi_bval_fname, hardi_bvec_fname)
+gtab = gradient_table(bvals, bvecs)
+
+"""
+To accelerate computation a brain mask must be computed. A median-otsu filter
+isolates the main signal and then, using a floodfill, the main 'blob' is
+picked. The resulting mask is saved for visual inspection.
+"""
+
+b0_indices = np.where(gtab.bvals == 0)[0]
+mean_b0 = data[..., b0_indices]
+mean_b0 = np.mean(mean_b0, axis=-1)
+_, mask = median_otsu(mean_b0)
+
+mask_labeled, _ = ndi.label(mask)
+unique, count = np.unique(mask_labeled, return_counts=True)
+val = unique[np.argmax(count[1:])+1]
+mask[mask_labeled != val] = 0
+
+save_nifti('mask.nii.gz', mask.astype(np.uint8), affine)
+
+"""
+Using a ResDNN for sh_order 8 (default) and load the appropriate weights.
+Fit the data and save the resulting fODF.
+"""
+
+resdnn_model = HistoResDNN(verbose=True)
+resdnn_model.fetch_default_weights()
+predicted_sh = resdnn_model.predict(data, gtab, mask=mask)
+
+save_nifti('predicted_sh.nii.gz', predicted_sh, affine)
+
+"""
+Preparing the scene using FURY. The ODF slicer expects spherical function
+and the mean spherical harmonic amplitude is used as background.
+The ODF slicer and the background image are added as actors and a mid-coronal
+slice is selected.
+"""
+
+interactive = False
+sphere = get_sphere('repulsion724')
+predicted_sf = sh_to_sf(sh=predicted_sh, sphere=sphere,
```

Comment:
There is a better way to do this here. As long as you provide the ``B_matrix`` in ``odf_slicer`` your can load directly the spherical harmonics (the spherical function is then saved as a memory map -- heavily reducing memory usage).